homeassistant/components/shelly/climate.py

```diff
@@ -316,5 +316,5 @@ def _handle_coordinator_update(self) -> None:
                 self.block = block
 
                 _LOGGER.debug("Entity %s attached to block", self.name)
-                self.async_write_ha_state()
-                return
```

Comment:
The return here is to break the loop when block is found, it is copied from:
https://github.com/home-assistant/core/blob/52e169db02d2ad762c2cc74e044fc04373219432/homeassistant/components/shelly/entity.py#L696-L713

But there we only attach one block. We should first find the two blocks than attach them and break the loop.

Reply:
Why do we need to break the loop if we only have two blocks here?

Reply:
You are correct about not having to break the loop however, I am not sure if it is good that we directly attach a block and what will happen if we somehow attach only one block